### PR TITLE
Unrevert "Create wmainmenubar controls in coreservices so they are available to controllers on startup"

### PR DESCRIPTION
**Pull request by ywwg**
Reverts mixxxdj/mixxx#4660,
Includes fix

### CODE REVIEW

File: src/coreservices.cpp [hunk 1]
```diff
@@ -423,6 +423,29 @@ void CoreServices::initialize(QApplication* pApp) {
...
+            ConfigKey("[Master]", "maximize_library"),
+            ConfigKey("[Samplers]", "show_samplers"),
+            ConfigKey("[EffectRack1]", "show"),
+            ConfigKey("[Skin]", "show_4effectunits"),
+            ConfigKey("[Master]", "show_mixer"),
+    };
+
+    m_uiControls.reserve(uiKeys.size());
+    for (const auto& key : uiKeys) {
+        m_uiControls.emplace_back(std::make_unique<ControlPushButton>(
```
Comment (daschuer) [inline src/coreservices.cpp:444]: persist = true already fetches the last value from the config.
The default value is used as a default if there is not yet anything stored.

https://github.com/mixxxdj/mixxx/blob/c7b4dbbb572930f8a5ae080aac71b99728c81966/src/control/control.cpp#L65

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/coreservices.cpp (hunk 1 above)
Comment (ywwg) [reply src/coreservices.cpp:444]: ah got it.  Fixed.

Comment (ywwg) [reply src/coreservices.cpp:444]: Hm, does that mean we need to pick the first-run defaults here?  Seems so

**Comment by ywwg** [comment]
Tested first-run case by moving ~/.mixxx aside, looks good

**Review by daschuer — APPROVED** [review]
Thank you.

File: src/coreservices.cpp [hunk 2]
```diff
@@ -423,6 +423,34 @@ void CoreServices::initialize(QApplication* pApp) {
...
+        bool persist;
+        bool defaultValue;
+    };
+    const std::vector<UIControlConfig> uiControls = {
+            {ConfigKey("[Master]", "skin_settings"), false, false},
+            {ConfigKey("[Microphone]", "show_microphone"), true, true},
+            {ConfigKey(VINYL_PREF_KEY, "show_vinylcontrol"), true, false},
+            {ConfigKey("[PreviewDeck]", "show_previewdeck"), true, true},
+            {ConfigKey("[Library]", "show_coverart"), true, true},
+            {ConfigKey("[Master]", "maximize_library"), true, false},
```
Comment (ronso0) [inline src/coreservices.cpp:441]:
```suggestion
            {ConfigKey("[Master]", "maximize_library"), false, false},
```
This is not persistent with official skins anyway.

File: src/coreservices.cpp [hunk 3]
```diff
@@ -423,6 +423,34 @@ void CoreServices::initialize(QApplication* pApp) {
...
+    struct UIControlConfig {
+        ConfigKey key;
+        bool persist;
+        bool defaultValue;
+    };
+    const std::vector<UIControlConfig> uiControls = {
+            {ConfigKey("[Master]", "skin_settings"), false, false},
+            {ConfigKey("[Microphone]", "show_microphone"), true, true},
+            {ConfigKey(VINYL_PREF_KEY, "show_vinylcontrol"), true, false},
+            {ConfigKey("[PreviewDeck]", "show_previewdeck"), true, true},
```
Comment (ronso0) [inline src/coreservices.cpp:439]:
```suggestion
            {ConfigKey("[PreviewDeck]", "show_previewdeck"), true, false},
```
hidden by default in all official skins.
Do we need to change that?

File: src/coreservices.cpp [hunk 4]
```diff
@@ -423,6 +423,34 @@ void CoreServices::initialize(QApplication* pApp) {
...
+    m_uiControls.clear();
+
+    struct UIControlConfig {
+        ConfigKey key;
+        bool persist;
+        bool defaultValue;
+    };
+    const std::vector<UIControlConfig> uiControls = {
+            {ConfigKey("[Master]", "skin_settings"), false, false},
+            {ConfigKey("[Microphone]", "show_microphone"), true, true},
```
Comment (ronso0) [inline src/coreservices.cpp:437]:
```suggestion
            {ConfigKey("[Microphone]", "show_microphone"), true, false},
```

File: src/coreservices.cpp [hunk 5]
```diff
@@ -423,6 +423,34 @@ void CoreServices::initialize(QApplication* pApp) {
...
+            {ConfigKey(VINYL_PREF_KEY, "show_vinylcontrol"), true, false},
+            {ConfigKey("[PreviewDeck]", "show_previewdeck"), true, true},
+            {ConfigKey("[Library]", "show_coverart"), true, true},
+            {ConfigKey("[Master]", "maximize_library"), true, false},
+            {ConfigKey("[Samplers]", "show_samplers"), true, true},
```
Comment (ronso0) [inline src/coreservices.cpp:442]: Hidden by default, shown in Deere.
Should we unify that to visible? (maybe)
If so, do we need a menu bar button? (if we add this and 'Show Effect Rack' the View men would be complete, wouldn't it?)

**Review by ronso0 — CHANGES_REQUESTED** [review]

**Comment by ronso0** [comment]
Ha! A few minutes too late...

**Comment by daschuer** [comment]
Oh Sorry. It could be a nice GitHub feature to see if there are unpublished reviews ...

**Comment by ronso0** [comment]
I could have sel-requested a review, but I only announced it in Zulip.

Anyway, we can discuss my review comments and based on the outcome I can open a follow-up PR.

**Pull request merged**